src/lua/LuaFileSystem.cpp

```diff
@@ -112,9 +180,9 @@ static int l_filesystem_read_dir(lua_State *l)
 		lua_setfield(l, -2, "mtime");
 
 		if (info.IsDir())
-			lua_rawseti(l, dirsTable, lua_rawlen(l, dirsTable) + 1);
+			lua_rawseti(l, dirsTable, (int)lua_rawlen(l, dirsTable) + 1);
 		else
-			lua_rawseti(l, filesTable, lua_rawlen(l, filesTable) + 1);
+			lua_rawseti(l, filesTable, (int)lua_rawlen(l, filesTable) + 1);
```

Comment:
Instead of querying `lua_rawlen` every loop iteration, this code should just keep a counter variable (e.g. `int i = 1;`) and increment it every time it appends to the lua table.

Reply:
Ahh yes, I just did the explicit cast there to remove an annoying compiler warning but didn't really look at the code.

Fixed :-)